src/phoenix/datasets/experiments.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import datetime
+from typing import (
+    Any,
+    Callable,
+    Coroutine,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    TypedDict,
+    Union,
+)
+
+import httpx
+
+from phoenix.config import (
+    get_env_collector_endpoint,
+    get_env_host,
+    get_env_port,
+)
+from phoenix.evals.executors import get_executor_on_sync_context
+from phoenix.evals.models.rate_limiters import RateLimiter
+
+JSONSerializable = Union[Dict[str, Any], List[Any], str, int, float, bool, None]
+ExperimentCallable = Callable[[JSONSerializable], JSONSerializable]
+ExperimentCoroutineFn = Callable[[JSONSerializable], Coroutine[Any, Any, JSONSerializable]]
```

Comment:
How about just `SyncTask` and `AsyncTask`?

Reply:
I was thinking that the type name should be more descriptive than "task" so it indicates what kinds of functions the user can pass in

Reply:
I think to Xander's point though I would groc it better if the type matched the language.

Reply:
sounds good, I'll use `ExperimentTask` and `AsyncExperimentTask` as to not clobber `asyncio.Task`